sklearn/decomposition/tests/test_rpca.py

```diff
@@ -0,0 +1,26 @@
+import numpy as np
+from sklearn.decomposition import RobustPCA
+from sklearn.utils.testing import assert_array_almost_equal
+
+
+def _gen_synthetic(n, density, rank):
+    r = rank  # Rank
+    X = np.random.normal(0, 1/float(n), size=(n, r))
+    Y = np.random.normal(0, 1/float(n), size=(n, r))
+    L = np.dot(X, Y.T)
+    p = density/2
+    S = np.random.choice([0, 1, -1], size=(n, n), p=[1 - 2*p, p, p])
+
+    return L, S
+
+
+def test_rpca():
+    rpca = RobustPCA()
+    rank = 25
+    L, S = _gen_synthetic(500, 0.05, rank)
+    X = L + S
+    rpca.fit(X)
+
+    assert rpca.n_components_ == rank
+    assert_array_almost_equal(L, rpca.low_rank_)
+    assert_array_almost_equal(L, rpca.inverse_transform(rpca.transform(X)))
```

Comment:
@vighneshbirodkar Not sure if I'm missing sth here, but shouldn't this line be
```
...almost_equal(X, ...
```
instead of `...almost_equal(L,...`? This test is supposed to test inverse transform indeed recovers X from transform's return values, right?

Reply:
Yes you are right.

Reply:
@vighneshbirodkar Can you fix it so this PR may be merged?

Reply:
Upon carefully looking I realized that none of these equalities make sense. `X` is a low rank matrix with some sparse noise. That's why projecting X directly onto the sub-space given by `rpca.components_` does not make sense. After calling `fit()` if we get new data, we would have to remove the sparse noise from it before projecting it onto the low dimensional sub-space.

@amueller I know it has been a long time but I'd like to know your thoughts.